harness/determined/pytorch/_pytorch_trial.py

```diff
@@ -788,16 +1108,38 @@ def _load(self, load_path: pathlib.Path) -> None:
                     "callback will be initialized from scratch"
                 )
 
-        # Load workload sequencer state.
-        wlsq_path = load_path.joinpath("workload_sequencer.pkl")
-        if self.wlsq is not None and wlsq_path.exists():
-            with wlsq_path.open("rb") as f:
-                self.wlsq.load_state(pickle.load(f))
+        save_path = load_path.joinpath("trial_state.pkl")
+        if save_path.exists():
+            with save_path.open("rb") as f:
+                self._load_state(pickle.load(f))
+        else:
+            # Support legacy save states
+            wlsq_path = load_path.joinpath("workload_sequencer.pkl")
+            if wlsq_path.exists():
+                with wlsq_path.open("rb") as f:
+                    self._load_state(pickle.load(f))
```

Comment:
I still don't see how this is going to work when the new TrialState looks like:
```python
class TrialState:
    def __init__(
            self,
            trial_id: int,
            last_ckpt: int = 0,
            steps_completed: int = 0,
            step_id: int = 0,
            last_val: int = 0,
            batches_trained: int = None,
            epochs_trained: int = None,
    ) -> None:
```

and the old SavableState looks like:
```python
class SavableState:
    def __init__(
        self,
        trial_id: int,
        last_ckpt: int = 0,
        steps_completed: int = 0,
        step_id: int = 0,
        last_val: int = 0,
    ) -> None:
```

It seems like you're going to have `batches_trained` and `epochs_trained` as None in this legacy codepath... is that expected?

Reply:
nope, they should be default 0. fixed.